### PR TITLE
[Structural] Removing the excessive number of registrations regarding the CL (in progress - to discuss)

### PR DESCRIPTION
**Pull request by AlejandroCornejo**
In this PR I'm removing all the registrations of the templated laws by instantiating the laws in the factory as suggested by @roigcarlo. If you come up with a better solution please feel free to share!

Thank you

### CODE REVIEW

File: applications/StructuralMechanicsApplication/structural_mechanics_application.cpp [hunk 1]
```diff
@@ -628,29 +628,29 @@ void KratosStructuralMechanicsApplication::Register() {
...
-    KRATOS_REGISTER_CONSTITUTIVE_LAW("SmallStrainIsotropicPlasticity3DVonMisesMohrCoulomb", mSmallStrainIsotropicPlasticity3DVonMisesMohrCoulomb);
-    KRATOS_REGISTER_CONSTITUTIVE_LAW("SmallStrainIsotropicPlasticity3DMohrCoulombVonMises", mSmallStrainIsotropicPlasticity3DMohrCoulombVonMises);
-    KRATOS_REGISTER_CONSTITUTIVE_LAW("SmallStrainIsotropicPlasticity3DMohrCoulombMohrCoulomb", mSmallStrainIsotropicPlasticity3DMohrCoulombMohrCoulomb);
-    KRATOS_REGISTER_CONSTITUTIVE_LAW("SmallStrainIsotropicPlasticity3DMohrCoulombDruckerPrager", mSmallStrainIsotropicPlasticity3DMohrCoulombDruckerPrager);
-    KRATOS_REGISTER_CONSTITUTIVE_LAW("SmallStrainIsotropicPlasticity3DMohrCoulombTresca", mSmallStrainIsotropicPlasticity3DMohrCoulombTresca);
-    KRATOS_REGISTER_CONSTITUTIVE_LAW("SmallStrainIsotropicPlasticity3DTrescaMohrCoulomb", mSmallStrainIsotropicPlasticity3DTrescaMohrCoulomb);
-    KRATOS_REGISTER_CONSTITUTIVE_LAW("SmallStrainIsotropicPlasticity3DDruckerPragerMohrCoulomb", mSmallStrainIsotropicPlasticity3DDruckerPragerMohrCoulomb);
+    //KRATOS_REGISTER_CONSTITUTIVE_LAW("SmallStrainIsotropicPlasticity3DVonMisesVonMises", mSmallStrainIsotropicPlasticity3DVonMisesVonMises);
```
Comment (loumalouomega) [inline applications/StructuralMechanicsApplication/structural_mechanics_application.cpp]: I would remove it in the final version

File: applications/StructuralMechanicsApplication/structural_mechanics_application.h [hunk 2]
```diff
@@ -482,29 +482,29 @@ class KRATOS_API(STRUCTURAL_MECHANICS_APPLICATION) KratosStructuralMechanicsAppl
...
-    const GenericSmallStrainIsotropicPlasticity <GenericConstitutiveLawIntegratorPlasticity<MohrCoulombYieldSurface<DruckerPragerPlasticPotential<6>>>> mSmallStrainIsotropicPlasticity3DMohrCoulombDruckerPrager;
-    const GenericSmallStrainIsotropicPlasticity <GenericConstitutiveLawIntegratorPlasticity<MohrCoulombYieldSurface<TrescaPlasticPotential<6>>>> mSmallStrainIsotropicPlasticity3DMohrCoulombTresca;
-    const GenericSmallStrainIsotropicPlasticity <GenericConstitutiveLawIntegratorPlasticity<TrescaYieldSurface<MohrCoulombPlasticPotential<6>>>> mSmallStrainIsotropicPlasticity3DTrescaMohrCoulomb;
-    const GenericSmallStrainIsotropicPlasticity <GenericConstitutiveLawIntegratorPlasticity<DruckerPragerYieldSurface<MohrCoulombPlasticPotential<6>>>> mSmallStrainIsotropicPlasticity3DDruckerPragerMohrCoulomb;
+    // const GenericSmallStrainIsotropicPlasticity <GenericConstitutiveLawIntegratorPlasticity<VonMisesYieldSurface<VonMisesPlasticPotential<6>>>> mSmallStrainIsotropicPlasticity3DVonMisesVonMises;
```
Comment (loumalouomega) [inline applications/StructuralMechanicsApplication/structural_mechanics_application.h]: Idem

**Review by loumalouomega — COMMENTED** [review]
The important question, does it work?

File: applications/StructuralMechanicsApplication/structural_mechanics_application.cpp (hunk 1 above)
Comment (AlejandroCornejo) [reply applications/StructuralMechanicsApplication/structural_mechanics_application.cpp]: of course, this is work in progress :)

**Comment by AlejandroCornejo** [comment]
> The important question, does it work?

Yes! it does

**Comment by loumalouomega** [comment]
Check the Travis

**Review by loumalouomega — CHANGES_REQUESTED** [review]
@roigcarlo says this is not what he proposed

**Pull request closed without merging**